crates/bevy_image/src/image.rs

```diff
@@ -1400,6 +1420,14 @@ impl Image {
                 let r = (r as f64 * u32::MAX as f64) as u32;
                 bytes[0..4].copy_from_slice(&u32::to_le_bytes(r));
             }
+            TextureFormat::R16Float => {
+                // Convert to grayscale with minimal loss if color is already gray
+                let linear = LinearRgba::from(color);
+                let luminance = Xyza::from(linear).y;
+                let [r, _, _, _] = LinearRgba::gray(luminance).to_f32_array();
```

Comment:
Why is this step needed? Couldn't luminance be used directly? Users may also expect R to be used here, but I can see why you chose G.

Reply:
Oh I see it was already done this way. ..I wonder why